src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -271,6 +276,68 @@ private static ImmutableArray<Diagnostic> EmitRegexMethod(IndentedTextWriter wri
                 writer.WriteLine($"            }}");
             }
 
+            if ((requiredHelpers & RequiredHelperFunctions.IsBoundary) != 0)
+            {
+                writer.WriteLine();
+                writer.WriteLine($"            /// <summary>Determines whether the character at the specified index is a boundary.</summary>");
+                writer.WriteLine($"            [global::System.Runtime.CompilerServices.MethodImpl(global::System.Runtime.CompilerServices.MethodImplOptions.AggressiveInlining)]");
+                writer.WriteLine($"            private static bool IsBoundary(global::System.ReadOnlySpan<char>inputSpan, int index, int startpos, int endpos)");
+                writer.WriteLine($"            {{");
+                writer.WriteLine($"                return (index > startpos && IsBoundaryWordChar(inputSpan![index - 1])) !=");
+                writer.WriteLine($"                       (index < endpos && IsBoundaryWordChar(inputSpan![index]));");
+                writer.WriteLine();
+                writer.WriteLine($"                bool IsBoundaryWordChar(char ch)");
+                writer.WriteLine($"                {{");
+                writer.WriteLine($"                    global::System.ReadOnlySpan<byte> asciiLookup = new byte[]");
+                writer.WriteLine($"                    {{");
+                writer.WriteLine($"                        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xFF, 0x03,");
+                writer.WriteLine($"                        0xFE, 0xFF, 0xFF, 0x87, 0xFE, 0xFF, 0xFF, 0x07");
+                writer.WriteLine($"                    }};");
```

Comment:
I thought of that too when I was writing it, but since we had said that the idea was eventually to remove them from here and instead expose them through RegexRunner I didn't end up doing the refactoring here.